infra/util/test_helper/MockCallback.hpp

```diff
@@ -106,6 +106,26 @@ namespace infra
             };
         }
     };
+
+    template<class T, class P1, class P2, class P3, std::size_t ExtraSize>
+    class VerifyingFunctionMock<T(P1, P2, P3), ExtraSize>
```

Comment:
Why isn't this a variadic template parameter for P1 P2 and P3?

Reply:
I'll look into this later